pkg/reconciler/identity/reconciler.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package identity
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/ptr"
+
+	"github.com/google/knative-gcp/pkg/duck"
+	"github.com/google/knative-gcp/pkg/reconciler/identity/resources"
+)
+
+type Identity struct {
+	// For dealing with Topics and Pullsubscriptions
+	KubeClient kubernetes.Interface
+}
+
+func (i *Identity) ReconcileWorkloadIdentity(ctx context.Context, projectID, namespace string, identifiable duck.Identifiable) (*corev1.ServiceAccount, error) {
```

Comment:
here is worth adding a clear comment. We are using this method from different places. It would be nice for other folks to just read the comment and understand what the method is doing.